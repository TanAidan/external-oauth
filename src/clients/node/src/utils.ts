import Types from './types'

/**
 * Helper to build a new URL from different params
 */
export function toURL(origin: string, baseURL: string, endpoint: string, queryString?: Types.RequestQueryString): URL {
  const removeLeadingSlash = (text: string) => {
    return text.replace(/^\//, '')
  }

  const removeTrailingSlash = (text: string) => {
    return text.replace(/\/$/, '')
  }

  const urlParts: string[] = []
  urlParts.push(removeTrailingSlash(origin))
  urlParts.push(removeLeadingSlash(removeTrailingSlash(baseURL)))
  urlParts.push(removeLeadingSlash(endpoint))

  const url = new URL(urlParts.join('/'))

  if (queryString) {
    Object.keys(queryString).forEach(key => url.searchParams.append(key, String(queryString[key])))
  }

  return url
}

/**
 * Helper to remove all undefined keys
 * @param obj {object}
 */
export function cleanHeaders(obj: Record<string, any>) {
  return Object.keys(obj).reduce((acc, key: string) => {
    if (obj[key] !== undefined) {
      acc[key] = obj[key]
    }
    return acc
  }, {} as Record<string, any>)
}
