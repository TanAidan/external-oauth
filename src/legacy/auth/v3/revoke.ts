import { asyncMiddleware } from '../../errorHandler'
import { TRevokeRequest } from './types'
// import { revokeAuthV3 } from '../../clients/integrations'
import { Response } from 'express'
import { MissingParameter } from '../../errors'

export const revoke = asyncMiddleware(async (req: TRevokeRequest, res: Response) => {
  // const { authId } = req.params
  // const { buid } = req
  const { clientId } = req.query

  if (!clientId) {
    throw new MissingParameter('clientId')
  }

  // await revokeAuthV3({
  //   clientId,
  //   buid,
  //   authId,
  //   servicesTableName: req.stageVariables.servicesTableId
  // })

  res.status(200).end()
})
