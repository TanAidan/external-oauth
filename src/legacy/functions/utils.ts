// Is the given object empty e.g. {}?
export function isEmpty(obj: Object) {
  return Object.getOwnPropertyNames(obj).length === 0
}
