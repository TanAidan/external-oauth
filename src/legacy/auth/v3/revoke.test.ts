import { MiddlewareTestHarness } from '../../../../tests/utils'
// import { revokeAuthV3 } from '../../clients/integrations'
import { revoke } from './revoke'

jest.mock('../../clients/integrations')

describe('revoke', () => {
  const path = 'test-buid/test-auth-id'

  const setup = () =>
    new MiddlewareTestHarness({
      configureRequest: (req: any) => {
        req.buid = 'test-buid'
      },
      testMiddleware: revoke,
      pathParams: ['buid', 'authId']
    })

  // it('revokes the authentication', async () => {
  //   await setup()
  //     .get(path)
  //     .query('clientId=test-client-id')
  //     .expect(200)

  //   expect(revokeAuthV3).toMatchSnapshot()
  // })

  it('raises an error if the clientId parameter is missing', async () => {
    const test = setup()

    await test.get(path).expect(400)

    expect(test.err).toMatchSnapshot()
  })
})
