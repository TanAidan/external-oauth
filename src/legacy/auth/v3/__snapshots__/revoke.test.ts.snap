// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`revoke raises an error if the clientId parameter is missing 1`] = `[MissingParameter: 'clientId' must be set]`;

exports[`revoke revokes the authentication 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "authId": "test-auth-id",
        "buid": "test-buid",
        "clientId": "test-client-id",
        "servicesTableName": "services-table-name",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
